File: src/main/java/no/nav/fo/veilarbregistrering/registrering/tilstand/RegistreringTilstandRepository.java
package no.nav.fo.veilarbregistrering.registrering.tilstand;

import java.util.List;
import java.util.Optional;

public interface RegistreringTilstandRepository {
    long lagre(RegistreringTilstand registreringTilstand);
    RegistreringTilstand oppdater(RegistreringTilstand registreringTilstand);
    RegistreringTilstand hentRegistreringTilstand(long id);
    List<RegistreringTilstand> finnRegistreringTilstanderMed(Status status);
    Optional<RegistreringTilstand> finnNesteRegistreringTilstandMed(Status status);
    int hentAntall(Status status);
    RegistreringTilstand hentTilstandFor(long registreringsId);
}

File: src/main/java/no/nav/fo/veilarbregistrering/registrering/tilstand/RegistreringTilstandRepository.kt
package no.nav.fo.veilarbregistrering.registrering.tilstand

interface RegistreringTilstandRepository {
    fun lagre(registreringTilstand: RegistreringTilstand): Long
    fun oppdater(registreringTilstand: RegistreringTilstand): RegistreringTilstand
    fun hentRegistreringTilstand(id: Long): RegistreringTilstand
    fun finnRegistreringTilstanderMed(status: Status): List<RegistreringTilstand>
    fun finnNesteRegistreringTilstandMed(status: Status): RegistreringTilstand?
    fun hentAntallPerStatus(): Map<String, Int>
    fun hentTilstandFor(registreringsId: Long): RegistreringTilstand
}